classes/article/ArticleGalleyDAO.inc.php

```diff
@@ -121,20 +129,23 @@ function getGalleysBySetting($settingName, $settingValue, $articleId = null, $jo
 	/**
 	 * @copydoc RepresentationDAO::getBySubmissionId()
 	 */
-	function getBySubmissionId($submissionId, $contextId = null) {
+	function getBySubmissionId($submissionId, $contextId = null, $submissionRevision = null) {
 		$params = array((int) $submissionId);
 		if ($contextId) $params[] = (int) $contextId;
+		if ($submissionRevision) $params[] = (int) $submissionRevision;
 
 		return new DAOResultFactory(
 			$this->retrieve(
-				'SELECT sf.*, g.*
+				'SELECT sf.*, g.*, sfg.version
 				FROM submission_galleys g
 				' . ($contextId?'INNER JOIN submissions s ON (g.submission_id = s.submission_id) ':'') . '
-				LEFT JOIN submission_files sf ON (g.file_id = sf.file_id)
-				LEFT JOIN submission_files nsf ON (nsf.file_id = g.file_id AND nsf.revision > sf.revision)
+				LEFT JOIN submission_galley_files sfg ON (sfg.galley_id = g.galley_id)
+				LEFT JOIN submission_files sf ON (sfg.file_id = sf.file_id)
+				LEFT JOIN submission_files nsf ON (nsf.file_id = sfg.file_id AND nsf.revision > sf.revision)
 				WHERE g.submission_id = ?
 					AND nsf.file_id IS NULL
 					' . ($contextId?' AND s.context_id = ? ':'') . '
+					' . ($submissionRevision?' AND sfg.version = ? ':'') . '
```

Comment:
I think the `. '` in the line before this line, as well as `'` in this line are not necessary?